conformance/conformance.proto

```diff
@@ -57,11 +57,13 @@ option java_package = "com.google.protobuf.conformance";
 //   2. parse the protobuf or JSON payload in "payload" (which may fail)
 //   3. if the parse succeeded, serialize the message in the requested format.
 message ConformanceRequest {
+  string test_name = 1;
```

Comment:
I don't think of it as an implementation detail -- the fact that each test has a name is a property of the testing framework. It's a public interface that you can specify a failure list of test names in a text file. I think having the test name could be useful to the testee in some cases. For example if a test is making the testee crash, they might want to log the test name in an exception handler.